ecosystem/typescript/sdk/SPEC.md

```diff
@@ -0,0 +1,220 @@
+# Aptos SDK Specification Document
+
+[Table of Content](#table-of-content)
+
+![Status](https://img.shields.io/badge/version-1.0-brightgreen.svg)
+
+## Overview
+
+The goal of this document is to set a shared standard for implementation and development of all Aptos SDKs.
+
+This is a living document, and should be changed and updated as changes are made when developer needs are discovered.
+
+### Requirement Prioritization
+
+The following document follows the [MoSCoW](https://en.wikipedia.org/wiki/MoSCoW_method) method of prioritising rules. Please follow the following guidelines when evaluating rules.
+
+- `MUST` - Rules labeled as **must** are requirements that should not be deviated from at any cost
+- `SHOULD` - Rules labeled as **should** are requirements that could be deviated from if needed, though this will have to be documented and cleared with all stakeholders before it can be disregarded.
+- `COULD` - Rules labeled as **could** are requirements that are desirable but not necessary and therefore would be nice to have where time and resources permit.
+
+We do not use the fourth **`won't`** level in this specification.
+
+## Table of Contents
+
+- Base Requirements
```

Comment:
do we want to expect an object or interaction layer independent of the actual API servers?

Reply:
that's a good question. not sure how deep we want to go with the implementation details. might be enough to say - have a [http client](https://github.com/aptos-labs/aptos-core/pull/8107/files#diff-66f82625c7f6d4c059797eef86f6d7be1de7d4fe3edc783009b480a0f4eb5137R188) and support [these](https://github.com/aptos-labs/aptos-core/pull/8107/files#diff-66f82625c7f6d4c059797eef86f6d7be1de7d4fe3edc783009b480a0f4eb5137R65) api endpoints. 

if they want it to be independent layers, it is up to them. One can create a client side sdk with a independent interaction layer, and one can create a node sdk where the interaction layer and endpoints are connected.